database/migrations/2024_08_07_234931_replace_eligible_courses_contexts_table.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+use App\Models\Course;
+use App\Models\Student;
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration {
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('eligible_context_courses', function (Blueprint $table) {
+            $table->id();
+            $table->foreignIdFor(Course::class);
+            $table->foreignIdFor(Student::class);
+            $table->timestamps();
+        });
+        DB::statement(<<<SQL
+            INSERT INTO eligible_context_courses (id, course_id, student_id, created_at, updated_at)
+            SELECT ECCC.id, ECCC.course_id, ECC.student_id, ECCC.created_at, ECCC.updated_at
+            FROM eligible_courses_context_courses AS ECCC
+            INNER JOIN eligible_courses_contexts AS ECC
+                ON ECC.id = ECCC.eligible_courses_major_id
```

Comment:
Using `eligible_courses_major_id` column here instead of `eligible_courses_context_id` because the foreign id column in `eligible_courses_context_courses` is based on [`EligibleCoursesMajor`](https://github.com/DoktorNik/course-companion/pull/31/files#diff-99b4ffec7d455b227b685f8f1a0d16bd6a0daea9d5bcf40b30511722aeb4afffL21) which seemed to be a typo reference.